R/mainTests.r

```diff
@@ -983,68 +984,31 @@ col_completed <- function(x, row, level_rows) {
 
 overcompleted_cols <- function(data, meta) {
   # ----------------------------------------------------------------------------------------------------------------------------------
-  # checking if region cols are completed in national rows
-
-  overcomplete_regional_cols <- function(matrixRow) {
-    # Filter the data to just the geographic levels that shouldn't have entries under region code and name
-    levels_incompatible_with_region <- c("National", "Local skills improvement plan area")
-
-    level_rows <- data %>% filter(geographic_level %in% levels_incompatible_with_region)
-
-    # Extract the columns for the geographic level that is being tested
-
-    cols <- matrixRow[2:4] %>% .[!is.na(.)]
-
-    # Apply over every column in the matrixRow (geographic_level) being tested
-
-    pre_output <- sapply(c(1:length(cols)), col_completed, row = matrixRow, level_rows = level_rows)
-
-    return(pre_output)
-  }
-
-  # ----------------------------------------------------------------------------------------------------------------------------------
-  # checking if local authority columns are completed for national, regional or mid-geography rows (ignoring LAD)
-
-  overcomplete_la_cols <- function(matrixRow) {
-    # This is a test that could benefit from more detail, and maybe a table in the error feedback
-
-    # Start by removing any geographic levels that are allowed to have entries in the LA ID and name columns
-    levels_compatible_with_la <- c("Local authority", "Local authority district", "School", "Provider", "Institution", "Planning area")
-
-    level_rows <- data %>% filter(!geographic_level %in% levels_compatible_with_la)
+  # Helper function for calculating overcomplete rows
+  # Feed in the geog level you want to test, and then the levels it's columns it is okay to be completed for
+  # It will return any columns that are detected as filled in for rows that they shouldn't be
+  #
+  # E.g. if you use geog_row = 4, that is the LAD row in the geog matrix...
```

Comment:
Do feel like it'd be good to switch to a data frame rather than a matrix at some point. Suspect it could make it more readable to use filter() and select() etc rather than using the matrix indices. Don't think it's worth the hassle adding it to a big PR like this, but would be tempted to look at trying it in a standalone PR.

Reply:
Yeah I think that's outside of this PR. There's some more of the code that would need updating as the current code needs it in a matrix to easily apply the function over the rows and columns (the reason why I used one to start with as it was the quickest way I could figure out to do what I want).

It's possible to do frame a dataframe but would need more rewriting than I'd like to do here.

Reply:
That's what I figured. I'll add it to the issues log in case we do want to pick it up at some point.